File: backend/app/models/relay.py
"""跨中心区域注册表模型（openspec: add-relay-gateway / relay-region-registry）。"""
from datetime import datetime

from sqlalchemy import Column, DateTime, Integer, String

from app.core.database import Base


class RelayGateway(Base):
    """区域注册表：一个路局中心一条记录。

    - code: 区域码（集群 region_code 挂接、fleet 组名与渲染文件名使用），创建后不可变
    - http_base_url / ssh_jump: 该局网关两腿路由；均为空 = 直连区域（如武清本地）
    - openresty_prefix: 该局网关 OpenResty 安装前缀（init/push 定位 conf/sbin；空 = 标准路径）
    - status: enabled / disabled（单局摘除，回退直连）
    """

    __tablename__ = "relay_gateways"

    id = Column(Integer, primary_key=True, index=True)
    code = Column(String(32), nullable=False, unique=True, index=True)
    name = Column(String(100), nullable=False)
    http_base_url = Column(String(255), nullable=True)
    ssh_jump = Column(String(255), nullable=True)
    openresty_prefix = Column(String(255), nullable=True)
    status = Column(String(16), nullable=False, default="enabled")
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

File: frontend/src/api/relay.ts
/**
 * 跨中心中继区域注册表 API（openspec: add-relay-gateway）。
 */
import api from '@/api/index'

export interface RelayGateway {
  id: number
  code: string
  name: string
  http_base_url: string | null
  ssh_jump: string | null
  openresty_prefix: string | null
  status: 'enabled' | 'disabled' | string
  created_at?: string | null
  updated_at?: string | null
}

export interface RelaySegment {
  name: string
  ok: boolean
  elapsed_ms?: number
  error?: string
  nodes?: { node: string; ok: boolean; error?: string }[]
}

export interface RelayHealthResult {
  region: string
  ok: boolean
  segments: RelaySegment[]
  error?: string
}

export interface RelayPushResult {
  ok: boolean
  region: string
  rc?: number
  status?: string
  hosts_pattern?: string
  listen_port?: number
}

export function listRelayGateways() {
  return api.get<RelayGateway[]>('/relay/gateways')
}

export function createRelayGateway(data: Partial<RelayGateway>) {
  return api.post<RelayGateway>('/relay/gateways', data)
}

export function updateRelayGateway(id: number, data: Partial<RelayGateway>) {
  return api.put<RelayGateway>(`/relay/gateways/${id}`, data)
}

export function deleteRelayGateway(id: number) {
  return api.delete(`/relay/gateways/${id}`)
}

export function pushRelayGatewayConfig(id: number) {
  return api.post<RelayPushResult>(`/relay/gateways/${id}/push-config`)
}

export function initRelayGateway(id: number) {
  return api.post<RelayPushResult>(`/relay/gateways/${id}/init`)
}

export function relayHealthCheck(region?: string) {
  const query = region ? `?region=${encodeURIComponent(region)}` : ''
  return api.get<RelayHealthResult | { regions: RelayHealthResult[] }>(`/relay/health-check${query}`)
}
